cluster-autoscaler/cloudprovider/ionoscloud/ionoscloud_cloud_provider.go

```diff
@@ -76,9 +76,6 @@ func (n *nodePool) IncreaseSize(delta int) error {
 		return err
 	}
 	targetSize := size + delta
-	if targetSize > n.max {
-		return fmt.Errorf("size increase exceeds upper bound of %d", n.max)
-	}
```

Comment:
I thought about this again. Removing this check is fine, but I guess it's not enough.

We saw this check fail when `targetSize == n.max + 1`.
Assuming that `n.max` and `delta` are alright, this means that `targetSize` is larger than it should be. Just removing this check here would just move the error to the Cloud API validations, as they would complain about exceeding the maximum.

We need to find out why `targetSize` has the wrong value. Here it's based on `GetNodeGroupSize()`. The included non-provider-specific size check uses something different: `TargetSize()` which in turn calls `GetNodeGroupTargetSize()`. They are different:
- `GetNodeGroupSize()` returns the current number of actually existing nodes in the node pool. 
- `GetNodeGroupTargetSize()` returns the node pool's specified (i.e. desired) node count.

As we're about to update the node pool's desired node count, we should use the latter. But we don't. I think that might be the root cause. So please also switch to `GetNodeGroupTargetSize()`.